app/models/document.rb

```diff
@@ -64,6 +64,13 @@ class Document < ApplicationRecord
   scope :active, ->() { where(archived: false) }
   scope :archived, ->() { where(archived: true) }
 
+  after_save do
+    if upload.present? && upload.blob.present? && upload.blob.filename.extension_without_delimiter.downcase == "heic" && !upload.blob.persisted?
```

Comment:
maybe `upload.filename.extension_without_delimiter.downcase` can be `Document#is_heic?`

Reply:
yes but with so many ampersands you'll be amazed